core/src/test/java/io/aiven/kafka/tieredstorage/ObjectKeyFactoryTest.java

```diff
@@ -0,0 +1,295 @@
+        assertThat(
+            objectKeyFactory.key(fields, REMOTE_LOG_SEGMENT_METADATA, ObjectKeyFactory.Suffix.TIME_INDEX).value()
+        ).isEqualTo("other/topic/7/file.timeindex");
+        assertThat(
+            objectKeyFactory.key(fields, REMOTE_LOG_SEGMENT_METADATA, ObjectKeyFactory.Suffix.PRODUCER_SNAPSHOT).value()
+        ).isEqualTo("other/topic/7/file.snapshot");
+        assertThat(
+            objectKeyFactory.key(fields, REMOTE_LOG_SEGMENT_METADATA, ObjectKeyFactory.Suffix.TXN_INDEX).value()
+        ).isEqualTo("other/topic/7/file.txnindex");
+        assertThat(
+            objectKeyFactory.key(fields, REMOTE_LOG_SEGMENT_METADATA, ObjectKeyFactory.Suffix.LEADER_EPOCH_CHECKPOINT)
+                .value()
+        ).isEqualTo("other/topic/7/file.leader-epoch-checkpoint");
+        assertThat(
+            objectKeyFactory.key(fields, REMOTE_LOG_SEGMENT_METADATA, ObjectKeyFactory.Suffix.MANIFEST).value()
+        ).isEqualTo("other/topic/7/file.rsm-manifest");
+    }
+
+    @Test
+    void nullPrefix() {
+        final ObjectKeyFactory objectKeyFactory = new ObjectKeyFactory(null, false);
+        assertThat(objectKeyFactory.key(REMOTE_LOG_SEGMENT_METADATA, ObjectKeyFactory.Suffix.LOG).value())
+            .isEqualTo(
+                "topic-AAAAAAAAAAAAAAAAAAAAAQ/7/00000000000000001234-AAAAAAAAAAAAAAAAAAAAAA.log");
+    }
+
+    @Test
+    void suffixForIndexTypes() {
+        assertThat(ObjectKeyFactory.Suffix.fromIndexType(RemoteStorageManager.IndexType.OFFSET))
+            .isEqualTo(ObjectKeyFactory.Suffix.OFFSET_INDEX)
+            .extracting("value")
+            .isEqualTo("index");
+        assertThat(ObjectKeyFactory.Suffix.fromIndexType(RemoteStorageManager.IndexType.TIMESTAMP))
+            .isEqualTo(ObjectKeyFactory.Suffix.TIME_INDEX)
+            .extracting("value")
+            .isEqualTo("timeindex");
+        assertThat(ObjectKeyFactory.Suffix.fromIndexType(RemoteStorageManager.IndexType.PRODUCER_SNAPSHOT))
+            .isEqualTo(ObjectKeyFactory.Suffix.PRODUCER_SNAPSHOT)
+            .extracting("value")
+            .isEqualTo("snapshot");
+        assertThat(ObjectKeyFactory.Suffix.fromIndexType(RemoteStorageManager.IndexType.TRANSACTION))
+            .isEqualTo(ObjectKeyFactory.Suffix.TXN_INDEX)
+            .extracting("value")
+            .isEqualTo("txnindex");
+        assertThat(ObjectKeyFactory.Suffix.fromIndexType(RemoteStorageManager.IndexType.LEADER_EPOCH))
+            .isEqualTo(ObjectKeyFactory.Suffix.LEADER_EPOCH_CHECKPOINT)
+            .extracting("value")
+            .isEqualTo("leader-epoch-checkpoint");
+    }
+
+    @Test
+    void prefixMasking() {
+        final ObjectKeyFactory objectKeyFactory = new ObjectKeyFactory("prefix/", true);
+        assertThat(objectKeyFactory.key(REMOTE_LOG_SEGMENT_METADATA, ObjectKeyFactory.Suffix.LOG).toString())
+            .isEqualTo("<prefix>/topic-AAAAAAAAAAAAAAAAAAAAAQ/7/00000000000000001234-AAAAAAAAAAAAAAAAAAAAAA.log");
+    }
```

Comment:
I guess
```suggestion
        assertThat(objectKeyFactory.key(REMOTE_LOG_SEGMENT_METADATA, ObjectKeyFactory.Suffix.LOG)
            .hasToString("<prefix>/topic-AAAAAAAAAAAAAAAAAAAAAQ/7/00000000000000001234-AAAAAAAAAAAAAAAAAAAAAA.log");
```
and same below.

Reply:
Thanks, done